src/classes/HH_OppContactRoles_TEST.cls

```diff
@@ -504,5 +504,143 @@ private class HH_OppContactRoles_TEST {
             system.assertEquals(0, listSoft[4]);
         }
     }
-       
+
+    /*********************************************************************************************************
+    * @description Tests that an Opportunity Contact Role is created when the deceased member is set as the primary
+    * @return void
+    */
+    private static testMethod void opportunityDeceasedPrimaryAccount() {
+        
+        npe01__Contacts_and_Orgs_Settings__c contactSettingsForTests = UTIL_CustomSettingsFacade.getContactsSettingsForTests(
+            new npe01__Contacts_and_Orgs_Settings__c (
+                npe01__Opportunity_Contact_Role_Default_role__c = 'Donor'));
+        
+        npo02__Households_Settings__c householdSettingsForTests = UTIL_CustomSettingsFacade.getHouseholdsSettingsForTests(
+            new npo02__Households_Settings__c (
+                npo02__Household_Contact_Roles_On__c = true,
+                npo02__Household_Member_Contact_Role__c='Household Member'));
+
+        UTIL_UnitTestData_TEST.createAccountContactTestData(CAO_Constants.HH_ACCOUNT_TYPE, 1, 1, 1);
+        Account acct = UTIL_UnitTestData_TEST.listAccT[0];
+        Contact con = UTIL_UnitTestData_TEST.listConT[0];        
+        con.Deceased__c = true;  
+        update con;
+                
+        //create opportunity
+        Opportunity o = new Opportunity(
+            Name='test oppty',
+            CloseDate=date.newInstance(2001,12,12),
+            StageName=UTIL_UnitTestData_TEST.getClosedWonStage(),
+            Amount=100,
+            AccountId=acct.Id,
+            Primary_Contact__c = con.Id
+        );
+        
+        Test.startTest();
+        insert o;
+        Test.stopTest();
+        //check for contact roles
+        List<OpportunityContactRole> createdRoles = [select Role, ContactId, IsPrimary from OpportunityContactRole where OpportunityId = :o.id order by ContactId];
+        System.assertEquals(1, createdRoles.size(), 'The deceased contact should be added as the primary Contact Role for the Opportunity.');
+        System.assertEquals(contactSettingsForTests.npe01__Opportunity_Contact_Role_Default_role__c, createdRoles[0].Role, 'The Role should match the default contact role.');
+        System.assert(createdRoles[0].IsPrimary, 'The Contact Role should be flagged as primary.');
+    }
+
+    /*******************************************************************************************************
+    * @description Move an Opportunity to a different HH Account and verifies primary OCR remains
+    */ 
+    static testmethod void testOppsAcctChangeHHToHH() {
```

Comment:
* I suggest to annotate test methods with `@IsTest` for better readability.
* Methods should use the `lowerCamelCase` naming convention. Please rename accordingly.